src/main/kotlin/study/Skill.kt

```diff
@@ -0,0 +1,22 @@
+package study
+
+sealed class Skill(val value: String) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Skill
+
+        if (value != other.value) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return value.hashCode()
+    }
+}
+
+class Hard(value: String) : Skill(value)
+
+class Soft(value: String) : Skill(value)
```

Comment:
```suggestion
sealed interface Skill {

    val value: String

    data class Soft(override val value: String) : Skill
    data class Hard(override val value: String) : Skill
}
```
data class, sealed interface를 활용하여 다음과 같이 개선해볼 수도 있습니다! data class를 활용하면 equals와 hashCode를 override 하지 않아도 됩니다.